option.go

```diff
@@ -166,6 +166,34 @@ func WithRequireConnectProtocolHeader() HandlerOption {
 	return &requireConnectProtocolHeaderOption{}
 }
 
+// WithConditionalHandlerOptions is a function that returns a HandlerOption.
+// It's used to conditionally apply HandlerOption to a Handler based on the Spec.
+// For example, to apply a HandlerOption only to a specific procedure:
+//
+//	connect.WithConditionalHandlerOptions(func(spec connect.Spec) (options []connect.HandlerOption) {
+//		if spec.Procedure == pingv1connect.PingServicePingProcedure {
+//			options = append(options, connect.WithReadMaxBytes(1024))
+//		}
+//		return options
+//	})
+func WithConditionalHandlerOptions(conditional func(spec Spec) []HandlerOption) HandlerOption {
+	return &conditionalHandlerOptions{conditional: conditional}
+}
+
+type conditionalHandlerOptions struct {
+	conditional func(spec Spec) []HandlerOption
+}
+
+func (o *conditionalHandlerOptions) applyToHandler(config *handlerConfig) {
```

Comment:
Please follow the ordering suggested by the style guide - in this case, `WithConditionalHandlerOptions` can stay here, but the `conditionalHandlerOptions` struct and methods on it go down below, with the others.